.github/workflows/ci.yml

```
@@ -14,35 +17,56 @@ jobs:
         uses: actions/checkout@v2
         with:
           lfs: true
-      - name: Setup .NET Core SDK 6.0
+      - name: Setup .NET 6 SDK
         uses: actions/setup-dotnet@v3
+        if: matrix.dotnet == '6.0'
         with:
           dotnet-version: 6.0.x
-      - name: Setup .NET Core SDK 7.0
+      - name: Setup .NET 7 SDK
         uses: actions/setup-dotnet@v3
         with:
           dotnet-version: 7.0.x
       - name: Restore
         run: dotnet restore
       - name: Build
-        run: dotnet build --no-restore
+        run: dotnet build --no-restore --framework net${{ matrix.dotnet }}
       - name: Test
-        run: dotnet test --collect "XPlat Code Coverage" --settings coverlet.runsettings --logger trx --no-build
+        run: dotnet test --collect "XPlat Code Coverage" --settings coverlet.runsettings --logger trx --framework net${{ matrix.dotnet }}
       - name: Upload coverage to Codecov
-        uses: codecov/codecov-action@v1
-        if: always()
-      - name: Publish Unit Test Results
-        uses: EnricoMi/publish-unit-test-result-action@v2
-        if: >
-          always() &&
-          github.event.sender.login != 'dependabot[bot]' &&
-          ( github.event_name != 'pull_request' || github.event.pull_request.head.repo.full_name == github.repository )
+        uses: codecov/codecov-action@v3
         with:
-          github_token: ${{ secrets.GITHUB_TOKEN }}
-          trx_files: TestResults/*.trx
+          flags: net${{ matrix.dotnet }}
+        if: always()
       - name: Upload Test Results
         if: always()
-        uses: actions/upload-artifact@v2
+        uses: actions/upload-artifact@v3
         with:
-          name: Unit Test Results
+          name: test_results_${{ matrix.dotnet }}.zip
           path: TestResults/*.trx
+
+  publish-test-results:
+    name: "Publish Tests Results"
+    needs: ci
+    runs-on: ubuntu-latest
+    permissions:
+      checks: write
+      pull-requests: write
+    if: always()
+    steps:
+      - name: Download Artifacts
+        uses: actions/download-artifact@v2
+        with:
+          path: artifacts
+      - name: Extract Artifacts
+        run: |
+          for file in artifacts/*.zip; do
+            if [[ -f "$file" ]]; then
+              dir="${file/%.zip/}"
```

Assuming this is `bash` and not some POSIX `sh`, stripping the suffix is more commonly done like so:
```suggestion
              dir="${file%.zip}"
```